lib/containers/common.pm

```diff
@@ -97,7 +97,18 @@ sub install_docker_when_needed {
     # docker daemon can be started
     systemctl('enable docker');
     systemctl('is-enabled docker');
-    systemctl('start docker');
+    # docker start, but taking bsc#1187479 into account. Please remove softfailure handling, once bsc#1187479 is solved.
+    if (systemctl('start docker', ignore_failure => 1) != 0) {
+        # Check for docker start timeout, bsc#1187479
+        if (script_run('journalctl -e | grep "timeout waiting for containerd to start"')) {
```

Comment:
This is wrong.  You want that the code under the if-condition gets executed when the grep text exists. So, `script_run` will return `0` in that case and the if block won't be executed. What you need is:


`if (script_run('journalctl -e | grep "timeout waiting for containerd to start"') == 0) {`

Reply:
https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/12795